contracts/AdapterStore.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+
+library MessengerTypes {
+    bytes32 public constant OFT_MESSENGER = bytes32("OFT_MESSENGER");
+    bytes32 public constant HYP_XERC20_ROUTER = bytes32("HYP_XERC20_ROUTER");
+}
+
+/**
+ * @dev A helper contract for chain adapters that support OFT or XERC20 messaging(via Hyperlane) on L1
+ * @dev Handles token => messenger/router mapping storage, as adapters are called via delegatecall and don't have relevant storage space
+ */
+contract AdapterStore is Ownable {
+    // (messengerType, l2ChainId, l1Token) => messenger address
+    mapping(bytes32 => mapping(uint256 => mapping(address => address))) public crossChainMessengers;
+
+    event MessengerSet(
+        bytes32 indexed messengerType,
+        uint256 indexed l2ChainId,
+        address indexed l1Token,
```

Comment:
Same comment for [l1Token](https://github.com/across-protocol/contracts/pull/914/files#r2012272807)